lumen/ai/agents.py

```diff
@@ -94,7 +94,8 @@ class Agent(Viewer, Actor):
 
     def __init__(self, **params):
         def _exception_handler(exception):
-            if str(exception) in self.interface.serialize()[-1]["content"]:
+            messages = self.interface.serialize()
+            if messages and str(exception) in messages[-1]["content"]:
```

Comment:
what's the magic number `-1` here? Will the exception always be in the last message? Can we be sure of this always?

Reply:
It's not always the last, but if it is, don't resend the exception.